ClientinfoPlugin.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace DNWS
+{
+  class ClientinfoPlugin : IPlugin
+  {
+    protected static Dictionary<String, int> clientinfoDictionary = null;
+    public ClientinfoPlugin()
+    {
+      if (clientinfoDictionary == null)
+      {
+        clientinfoDictionary = new Dictionary<String, int>();
+
+      }
+    }
+
+    public void PreProcessing(HTTPRequest request)
+    {
+      if (clientinfoDictionary.ContainsKey(request.Url))
```

Comment:
This is clientinfo plugin, you don’t need the code to track user request (as is in Stat plugin)